### DISCUSSION TITLE
getBlob returning broken image

### DISCUSSION THREAD

**Discussion by JayIsPainting** [bluesky-social/atproto#1843] [How to do it?] [answered]
Continuing on from this https://github.com/bluesky-social/atproto/discussions/1840

I now have my code properly posting to the point where I am getting a getBlob response  for all my posts.

https://bsky.social/xrpc/com.atproto.sync.getBlob?did=did:plc:vo37i5ooqwooxgkmbc3277kt&cid=bafkreiagmyiuh3xc73zuxbsodrtu547j7zcdfufbgutpcolf7cez5dgpba

but what I get back is a broken image.
`<img style="display: block;-webkit-user-select: none;margin: auto;background-color: hsl(0, 0%, 90%);transition: background-color 300ms;" src="https://bsky.social/xrpc/com.atproto.sync.getBlob?did=did:plc:vo37i5ooqwooxgkmbc3277kt&amp;cid=bafkreiglzozwphua6jngmlupuyzoblu7d2a3iqz44s755ntaclmlnqzyje">
`
I thought my CFML/JAVA code was doing something to the post data so I switched to curl but this produces the same result. 

`curl https://bsky.social/xrpc/com.atproto.repo.uploadBlob -F "file=@E:\test.jpg"  --header "Authorization:Bearer [redacted]" --header "Content-Type:image/jpg"`

Seems like it should be working but yet...

**Comment by JayIsPainting** [comment]
My createRecord json is:

{"repo":"did:plc:vo37i5ooqwooxgkmbc3277kt",
	"record":
		{"embed":
			{"images":
				[{
				"image":
					{
					"ref":
						{
						"$link":"bafkreiglzozwphua6jngmlupuyzoblu7d2a3iqz44s755ntaclmlnqzyje"
						},
						"size":30795,
						"$type":"blob",
						"mimeType":"image/jpg"
						},
				"alt":"brief alt text description of the image"
			}],
		"$type":"app.bsky.embed.images"
		},
	"text":"Image API Test posted at {ts '2023-11-10 09:46:19'}",
	"createdAt":"2023-11-10T09:46:19-07:00",
	"langs":["en-US"],
	"type":"app.bsky.feed.post"
	},
	"collection":"app.bsky.feed.post"}

**Reply by DavidBuchanan314** [reply]
It looks like you're trying to upload the file as if it were a form field, which is incorrect.

`curl -d "@E:\test.jpg" [rest of options here]` will probably do what you want

**Reply by JayIsPainting** [reply]
That throws {"error":"InternalServerError","message":"Internal Server Error"}

**Reply by JayIsPainting** [reply]
fwiw -d doesn't but - -data-raw works. But the image is still broken.

**Reply by JayIsPainting** [reply, marked as answer]
I was unable to get curl to work so I went back to using CFML and got it to work. (yay!)

it was all about how the image was being passed

local.imageFile = "E:\test.jpg">
local.imageFile = filereadbinary(local.imageFile) //convert to binary 

and then in cfhttp use type="body"

cfhttpparam(type="body" value=local.imageFile)

**Comment by katef** [comment]
Hi, I hit the same. with curl you want `--data-binary` not `--data-raw`.

I don't know what the difference is, except one works and the other doesn't.

**Reply by jay-elizaga-dev** [reply]
From the man page of curl: 

> [--data-raw](https://curl.se/docs/manpage.html#--data-raw) is almost the same but does not have a special interpretation of the @ character. To post data purely binary, you should instead use the [--data-binary](https://curl.se/docs/manpage.html#--data-binary) option.

https://curl.se/docs/manpage.html
